app/javascript/components/workflows/student/exams/show/lockdown/listeners.ts

```diff
@@ -40,6 +40,18 @@ const listeners: {
       detected('tried context menu', e);
     },
   },
+  {
+    event: 'beforeunload',
+    handler: (detected) => (e: Event): void => {
+      detected('tried to navigate away', e);
+    },
+  },
+  {
+    event: 'popstate',
```

Comment:
Do you want to detect `pushstate` as well here?

Reply:
It seems to me like you want to do this differently: I think you may want an `onbeforeunload` handler that offers students a dialog "Are you sure you want to navigate away?  If you do, your exam will be finalized and you cannot work on it any further!".  It's probably too easy to navigate unintentionally: if you thought you had focus in a text box but didn't, and press Backspace, Firefox will navigate away...

Reply:
Yeah, probably. I will add it.